core/oscillations.c

```diff
@@ -0,0 +1,220 @@
+/******************************************************************************
+ *                                                                            *
+ * OSCILLATIONS.C                                                             *
+ *                                                                            *
+ * Routines for neutrino oscillations                                         *
+ *                                                                            *
+ ******************************************************************************/
+#include "decs.h"
+
+#if RADIATION == RADTYPE_NEUTRINOS
+#if LOCAL_ANGULAR_DISTRIBUTIONS
+
+double get_dt_oscillations() {
+  set_Rmunu(); // So we have Nsph and nph
+  double nph_max = 0;
+#pragma omp parallel for reduction(max : nph_max) collapse(3)
+  ZLOOP {
+    nph_max = MY_MAX(nph_max, nph[i][j][k]); // 1/cm^3
+  }
+  // seconds
+  double dt_osc = 1. / (NUFERM * nph_max + SMALL);
+  dt_osc /= T_unit; // code units
+  dt_osc        = mpi_min(dt_osc);
+  return dt_osc;
+}
+
+void accumulate_local_angles() {
+  static const int LOCAL_ANGLES_SIZE = LOCAL_NUM_BASES * LOCAL_ANGLES_NX1 *
+                                       LOCAL_ANGLES_NX2 * LOCAL_ANGLES_NMU *
+                                       RAD_NUM_TYPES;
+  static const int LOCAL_STDDEV_SIZE =
+      LOCAL_NUM_BASES * LOCAL_ANGLES_NX1 * LOCAL_ANGLES_NX2 * LOCAL_ANGLES_NMU;
+
+  memset(local_angles, 0, LOCAL_ANGLES_SIZE * sizeof(double));
+  memset(local_stddev, 0, LOCAL_STDDEV_SIZE * sizeof(double));
+
+#pragma omp parallel
+  {
+    struct of_photon *ph = photon_lists[omp_get_thread_num()];
+    while (ph != NULL) {
+      if (ph->type != TYPE_TRACER) {
+        int ix1, ix2, icosth1, icosth2;
+        get_local_angle_bins(ph, &ix1, &ix2, &icosth1, &icosth2);
+        for (int b = 0; b < LOCAL_NUM_BASES; ++b) {
+#pragma omp atomic
+          local_angles[b][ix1][ix2][ph->type][icosth1] += ph->w;
+#pragma omp atomic
+          local_stddev[b][ix1][ix2][icosth1] += 1.;
```

Comment:
Used as a switch so we don't detect spurious crossings. Currently we're not outputting this to disk, but we could add it to the list of diagnostics.